src/procedures/proc_functions.c

```
@@ -0,0 +1,165 @@
+	if(array_len((SIValue *)args) != 0) return PROCEDURE_ERR;
+
+	ProcFunctionsPrivateData *pdata = rm_malloc(sizeof(ProcFunctionsPrivateData));
+	memset(pdata, 0, sizeof(ProcFunctionsPrivateData));
+
+	// initialize an iterator to the rax that contains all functions
+	rax *functions = __aeRegisteredFuncs;
+	raxStart(&pdata->iter, functions);
+	raxSeek(&pdata->iter, "^", NULL, 0);
+	pdata->output = array_new(SIValue, 4);
+	_process_yield(pdata, yield);
+
```

cppcheck

[error] Memory leak

(memleak)